### PR TITLE
Expose user Information in test app

### PR DESCRIPTION
**Pull request by dtrabo**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Add a button to auth section to opening new page with User Information

## Related PRs or issues

[AB#62366](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/62366)

### CODE REVIEW

File: Sasquatch/Sasquatch/ViewControllers/MSMainViewController.swift [hunk 1]
```diff
@@ -103,6 +107,16 @@ class MSMainViewController: UITableViewController, AppCenterProtocol {
...
     // Make sure the UITabBarController does not cut off the last cell.
     self.edgesForExtendedLayout = []
+    
+    if (isUserSignedIn) {
+      authInfoTVCell.isUserInteractionEnabled = true
+      authInfoLabel.text = "User authenticated"
+      authInfoLabel.isEnabled = true
+    } else {
+      authInfoTVCell.isUserInteractionEnabled = false
+      authInfoLabel.text = "User is no authenticated"
```
Comment (Jamminroot) [inline Sasquatch/Sasquatch/ViewControllers/MSMainViewController.swift]:
```suggestion
      authInfoLabel.text = "User is not authenticated"
```

File: Sasquatch/SasquatchSwift/AppCenterDelegateSwift.swift [hunk 2]
```diff
@@ -198,7 +198,7 @@ class AppCenterDelegateSwift: AppCenterDelegate {
   }
 
   // MSAuth section.
-  func signIn() {
+  func signIn( completionHandler: @escaping (_ signInInformation:MSUserInformation) -> Void) {
```
Comment (MatkovIvan) [inline Sasquatch/SasquatchSwift/AppCenterDelegateSwift.swift]: You have to update all `AppCenterDelegate` implementations

File: Sasquatch/Sasquatch/ViewControllers/MSMainViewController.swift [hunk 3]
```diff
@@ -103,6 +107,16 @@ class MSMainViewController: UITableViewController, AppCenterProtocol {
 
     // Make sure the UITabBarController does not cut off the last cell.
     self.edgesForExtendedLayout = []
+    
+    if (isUserSignedIn) {
+      authInfoTVCell.isUserInteractionEnabled = true
```
Comment (annakocheshkova) [inline Sasquatch/Sasquatch/ViewControllers/MSMainViewController.swift]: TVCell? What is TVCell? Is it on TV? 📺

Comment (jaelim-ms) [reply Sasquatch/Sasquatch/ViewControllers/MSMainViewController.swift]: It stands for TableViewCell I guess. 😝

Comment (dtrabo) [reply Sasquatch/Sasquatch/ViewControllers/MSMainViewController.swift]: Sorry, it's just interim naming :sweat_smile:

File: Sasquatch/Sasquatch/ViewControllers/MSAuthInfoViewController.swift [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+// Licensed under the MIT License.
+
+import UIKit
+
+class MSAuthInfoViewController: UITableViewController {
+  
+  @IBOutlet weak var accountId: UILabel!
+  @IBOutlet weak var accessToken: UILabel!
+  @IBOutlet weak var idToken: UILabel!
+
```
Comment (guperrot) [inline Sasquatch/Sasquatch/ViewControllers/MSAuthInfoViewController.swift]:
```suggestion
```

File: Sasquatch/Sasquatch/ViewControllers/MSAuthInfoViewController.swift [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    idToken.text = "ID Token:\n" + idTokenDecoded
+  }
+  
+  @IBAction func dismiss(_ sender: Any) {
+    self.dismiss(animated: true, completion: nil)
+  }
+  
+  override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+    if segue.destination is MSMainViewController {
+      let vc = segue.destination as? MSMainViewController
```
Comment (annakocheshkova) [inline Sasquatch/Sasquatch/ViewControllers/MSAuthInfoViewController.swift]: vc....

File: Sasquatch/Sasquatch/ViewControllers/MSMainViewController.swift [hunk 6]
```diff
@@ -116,17 +119,33 @@ class MSMainViewController: UITableViewController, AppCenterProtocol {
...
     self.appCenterEnabledSwitch.isOn = appCenter.isAppCenterEnabled()
     self.pushEnabledSwitch.isOn = appCenter.isPushEnabled()
     self.authSwitch.isOn = appCenter.isAuthEnabled()
+    if (self.userInformation != nil) {
+      authInfoCell.isUserInteractionEnabled = true
+      authInfoLabel.text = "User authenticated"
+      authInfoLabel.isEnabled = true
+    } else {
+      authInfoCell.isUserInteractionEnabled = false
+      authInfoLabel.text = "User is no authenticated"
```
Comment (annakocheshkova) [inline Sasquatch/Sasquatch/ViewControllers/MSMainViewController.swift]:
```suggestion
      authInfoLabel.text = "User is not authenticated"
```

File: Sasquatch/Sasquatch/ViewControllers/MSMainViewController.swift [hunk 7]
```diff
@@ -116,17 +119,33 @@ class MSMainViewController: UITableViewController, AppCenterProtocol {
...
+    updateViewState()
   }
   
   func updateViewState() {
     self.appCenterEnabledSwitch.isOn = appCenter.isAppCenterEnabled()
     self.pushEnabledSwitch.isOn = appCenter.isPushEnabled()
     self.authSwitch.isOn = appCenter.isAuthEnabled()
+    if (self.userInformation != nil) {
+      authInfoCell.isUserInteractionEnabled = true
+      authInfoLabel.text = "User authenticated"
```
Comment (annakocheshkova) [inline Sasquatch/Sasquatch/ViewControllers/MSMainViewController.swift]:
```suggestion
      authInfoLabel.text = "User is authenticated"
```

File: Sasquatch/Sasquatch/ViewControllers/MSAuthInfoViewController.swift [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    var idTokenDecoded: String = "None"
+    do {
+      let jsonDataAccess = try JSONSerialization.data(withJSONObject: decode(jwtToken: userInformation?.accessToken ?? "None"), options: .prettyPrinted)
+      let jsonDataId = try JSONSerialization.data(withJSONObject: decode(jwtToken: userInformation?.idToken ?? "None"), options: .prettyPrinted)
+      
+      accessTokenDecoded = String.init(data: jsonDataAccess, encoding: .utf8) ?? "None"
+      idTokenDecoded = String.init(data: jsonDataId, encoding: .utf8) ?? "None"
+      
+    } catch {
+      print(error.localizedDescription)
```
[the remark this answers is not in the document]
Comment (dtrabo) [reply Sasquatch/Sasquatch/ViewControllers/MSAuthInfoViewController.swift]: Actually `NSLog` you mean

**Pull request merged**